source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -534,24 +547,33 @@ void DnsMessageParser::buildResponseBuffer(DnsQueryContextPtr& query_context,
     ++serialized_queries;
     total_buffer_size += query_buffer.length();
 
-    for (const auto& answer : query_context->answers_) {
+    // Generate a set of random indices of the answer records that we serialize. We do this
+    // so that we don't always return the same records in the same order for every query
+    absl::flat_hash_set<size_t> indices{};
```

Comment:
I think using an absl::FixedArray would be more efficient here and then passing by reference?

Reply:
Let me take a look at this.  It does make sense.